x-pack/legacy/plugins/siem/public/pages/case/components/case_view/index.tsx

```diff
@@ -63,92 +60,64 @@ const BackgroundWrapper = styled.div`
 export interface CaseProps {
   caseId: string;
   initialData: Case;
-  isLoading: boolean;
 }
 
-export const CaseComponent = React.memo<CaseProps>(({ caseId, initialData, isLoading }) => {
-  const [{ data }, dispatchUpdateCaseProperty] = useUpdateCase(caseId, initialData);
-  const [isEditDescription, setIsEditDescription] = useState(false);
-  const [isEditTags, setIsEditTags] = useState(false);
-  const [isCaseOpen, setIsCaseOpen] = useState(data.state === 'open');
-  const [description, setDescription] = useState(data.description);
-  const [title, setTitle] = useState(data.title);
-  const [tags, setTags] = useState(data.tags);
+export const CaseComponent = React.memo<CaseProps>(({ caseId, initialData }) => {
+  const [{ data, isLoading, updateKey }, dispatchUpdateCaseProperty] = useUpdateCase(
+    caseId,
+    initialData
+  );
 
   const onUpdateField = useCallback(
-    async (updateKey: keyof Case, updateValue: string | string[]) => {
-      switch (updateKey) {
+    (newUpdateKey: keyof Case, updateValue: Case[keyof Case]) => {
+      switch (newUpdateKey) {
         case 'title':
-          if (updateValue.length > 0) {
+          const titleUpdate = getTypedPayload<string>(updateValue);
+          if (titleUpdate.length > 0) {
             dispatchUpdateCaseProperty({
               updateKey: 'title',
-              updateValue,
+              updateValue: titleUpdate,
             });
           }
           break;
         case 'description':
-          if (updateValue.length > 0) {
+          const descriptionUpdate = getTypedPayload<string>(updateValue);
+          if (descriptionUpdate.length > 0) {
             dispatchUpdateCaseProperty({
               updateKey: 'description',
-              updateValue,
+              updateValue: descriptionUpdate,
             });
-            setIsEditDescription(false);
           }
           break;
         case 'tags':
-          setTags(updateValue as string[]);
-          if (updateValue.length > 0) {
+          const tagsUpdate = getTypedPayload<string[]>(updateValue);
+          dispatchUpdateCaseProperty({
+            updateKey: 'tags',
+            updateValue: tagsUpdate,
+          });
+          break;
+        case 'state':
+          const stateUpdate = getTypedPayload<string>(updateValue);
+          if (data.state !== updateValue) {
             dispatchUpdateCaseProperty({
-              updateKey: 'tags',
-              updateValue,
+              updateKey: 'state',
+              updateValue: stateUpdate,
             });
-            setIsEditTags(false);
           }
-          break;
         default:
           return null;
       }
     },
-    [dispatchUpdateCaseProperty, title]
+    [dispatchUpdateCaseProperty, data.title]
```

Comment:
I do not think that you use data.title inside of your useCallback

Reply:
they get done on their component level, `UserActionTree`